doc/design/context.md

```diff
@@ -0,0 +1,171 @@
+## Context Design
+
+`Net` is the container and controller of a set of `Operator`. Each `Operator` in `Net` has a method called `Run` to make computation. The `Run` method of `Operator` is defined as follows:
+
+```c++
+Error Operator::Run(OpContext* context);
+```
+
+And the `OpContext` is defined as follows:
+
+```c++
+Struct OpContext {
+  Scope* scope;
+  DeviceContext* device_context;
+};
+```
+
+The `Run` method will take `Variable` (containing `Tensor`) from `Scope` to make computation on certain `DeviceContext`. `DeviceContext` provides necessary runtime resources for computation, including CudaStream, cublasHandle and so on.
```

Comment:
At now, we may not need DAGNet. 

But the ```Run``` method is actually executed on a specific CUDA stream. I think pass a CUDA stream to Operator is more flexible.